numba_cuda/numba/cuda/cudadrv/linkable_code.py

```diff
@@ -10,13 +11,19 @@ class LinkableCode:
     """
 
     def __init__(self, data, name=None):
-        self.data = data
         self._name = name
+        self._data = data
 
     @property
     def name(self):
         return self._name or self.default_name
 
+    @property
+    def data(self):
+        if isinstance(self._data, io.StringIO):
```

Comment:
This should be documented so that users are aware they can use a StringIO instance for data to generate the implementation on-the-fly

Reply:
Done in 0c85778